infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -292,13 +298,24 @@ private void rangeWrite(CorfuPayloadMsg<RangeWriteMsg> msg,
     }
 
     /**
-     * Resets the log unit server.
+     * Resets the log unit server via the BatchWriter.
      * Warning: Clears all data.
      */
     @ServerHandler(type = CorfuMsgType.RESET_LOGUNIT)
-    private void resetLogUnit(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        streamLog.reset();
-        dataCache.invalidateAll();
+    private synchronized void resetLogUnit(CorfuPayloadMsg<Long> msg,
```

Comment:
Why is this synchronized?

Reply:
To synchronize multiple reset requests coming in at the same time.